source/game.person.bmx

```diff
@@ -849,7 +849,8 @@ Type TPersonPersonalityData Extends TPersonPersonalityBaseData
 			dayOfBirth = _FixDate(dayOfBirth)
 		EndIf
 
-		Return GetWorldTime().GetTimeGoneFromString(dayOfBirth)
+		Local parts:String[] = dayOfBirth.split("-")
+		Return GetWorldTime().MakeRealTime(Int(parts[0]),Int(parts[1]),Int(parts[2]))
```

Comment:
Hier stimmen meines Erachtens beide Werte noch nicht:

now = "2022-01-27"
GetWorldTime().GetTimeGoneFromString(now) = 2098742400000
GetWorldTime().GetFormattedDate(2098742400000) = 00:00 01.04.2024
GetWorldTime().MakeRealTime(parts) = 2096484480000
GetWorldTime().GetFormattedDate(2096484480000) = 00:00 01.04.2024

Das schaue ich mir aber nochmal an und melde mich hier im PR.

Reply:
Ich glaube, das Problem mit GetTimeGoneFromString ist, dass es bei fehlender mitgegebener Urzeit immer eine Uhrzeit ergänzt (nämlich 0 Uhr). D.h. obwohl der 27.1. um 22 Uhr wäre, macht es dann einen Zeitstempel am selben Tag draus, der als "Realzeitdatum" dann aber eben nicht mehr der 27.1. ist.

Deshalb ja auch die Umstellung auf MakeRealTime.

Reply:
Die Methodennamen sind vielleicht etwas unguenstig gewaehlt.

MakeRealTime -> Generiere eine "time gone" basierend auf einer "echten" (360 Tage im Jahr, 12 Monate, ...) Zeitangabe.

MakeTime -> Generiere iene "time gone" basierend auf "Spielzeiten" (hier: 12 Tage im Jahr)


"MakeRealTime" nutzt man also, wenn man ein normales Kalenderdatum "umrechnen" will (in die Worldtime - die ja bspweise hier nur 12 Tage pro Jahr hat) - wie Geburtstage.

"MakeTime" nutzt man, wenn die Zahlen bereits zum "WorldTime"-Universum passen.


Wenn man eine Sportliga also am "13. März 1992" starten lassen will: `MakeRealTime(1992, 3, 13)`.
Soll sie hingegen am 14. Spieltag des Spiels um 5 Uhr starten, `MakeTime(GetStartYear(), 14, 5,0)` 

Ersteres hat ja mehrere "Märztage" an einem Spieltag, eine Uhrzeit anzugeben waere da moeglich (wuerde sich am Ende aber irgendwo nur in einigen "ingame-Minuten" widerspiegeln. Genauer waere es dann, mit dieser Zeit im Schlepptau den Spieltag zu ermitteln - und dann MakeTime() zu nutzen - mit eben der gewuenschten Uhrzeit:
```
Local realDateToWorldTime:Long = GetWorldTime().MakeRealTime(1992,3,13)
Local gameDay:Long = GetWorldTime().GetDay(realDateToWorldTime)
Local gameTime:Long = GetWorldTime().MakeTime(0, gameDay, 5, 0) '5 Uhr am Spieltag an dem der 13.3.1992 "stattfindet
```

Wenn Du Ideen hast, wie man das besser nennt - bzw in Kommentaren erklaert, gerne einen PR verfassen.

Reply:
MakeRealTime -> GetTimeGoneForRealDate
MakeTime -> GetTimeGoneForGameTime

Was bedeutet das jetzt aber für diesen PR?

Reply:
Das Du die Umbenennung gern gleich noch mit hier erledigen kannst :)

Und danach mergen.

Reply:
OK, aber nicht mehr heute.

Reply:
Eilt ja nicht.

Reply:
Umbenennung ist vorgenommen